packages/application-shell/src/components/login/login.js

```diff
@@ -98,155 +92,143 @@ export class Login extends React.PureComponent {
+                  isDisabled={formikProps.isSubmitting}
+                />
+                <TextField
+                  name="password"
+                  title={this.props.intl.formatMessage(messages.password)}
+                  isRequired={true}
+                  value={formikProps.values.password}
+                  touched={formikProps.touched.password}
+                  errors={formikProps.errors.password}
+                  onChange={formikProps.handleChange}
+                  onBlur={formikProps.handleBlur}
+                  isDisabled={formikProps.isSubmitting}
+                />
+                <Spacings.Inline>
+                  {/* FIXME: the <PrimaryButton> current does not behave like a
+                  "submit" button, making it impossible to handle "submit on enter".
+                  Once that is fixed in the underlying component, the "submit on enter"
+                  will work. */}
+                  <PrimaryButton
+                    label={
+                      this.state.loading
+                        ? this.props.intl.formatMessage(messages.validating)
+                        : this.props.intl.formatMessage(messages.signin)
+                    }
+                    onClick={formikProps.handleSubmit}
+                    isDisabled={formikProps.isSubmitting}
                   />
-                </Spacings.Stack>
-              </Spacings.Stack>
+                </Spacings.Inline>
 
-              <div>
-                <PrimaryButton
-                  label={
-                    this.state.loading
-                      ? this.props.intl.formatMessage(messages.validating)
-                      : this.props.intl.formatMessage(messages.signin)
+                <Countdown
+                  redirectTo={() =>
+                    redirect(`${this.props.adminCenterUrl}/reset-password`)
                   }
-                  onClick={this.handleSubmit}
-                />
-              </div>
-
-              <Countdown
-                redirectTo={() =>
-                  this.props.redirectTo(
-                    `${this.props.adminCenterUrl}/reset-password`
-                  )
-                }
-                isDisabled={this.state.loading}
-              >
-                {({ handleClick }) => (
-                  <a
-                    className={classnames(styles['forgot-pw'], {
-                      [styles.disabled]: this.state.loading,
-                    })}
-                    data-track-component="ForgotPassword"
-                    data-track-event="click"
-                    onClick={handleClick}
-                  >
-                    <FormattedMessage {...messages.forgotPassword} />
-                  </a>
-                )}
-              </Countdown>
-            </Spacings.Stack>
-          </fieldset>
-        </form>
-      </LoginBox>
+                  isDisabled={this.state.loading}
+                >
+                  {({ handleClick }) => (
```

Comment:
Nitpick: can we `handleClick: skipCountdown` to know what it does?

Reply:
Hmm the click is to show the dialog with the countdown. What do you mean by "skipCountdown"?